common/browserapp.cpp

```diff
@@ -28,20 +29,12 @@ bool BrowserApp::captivePortal()
 
 QString BrowserApp::profileName()
 {
-    static QString profileName = QStringLiteral("mozembed");
-    static bool argsChecked = false;
-
-    if (!argsChecked) {
-        const QStringList &arguments = QCoreApplication::arguments();
-
-        if (arguments.contains(QStringLiteral("-profile"))) {
-            int index = arguments.indexOf(QStringLiteral("-profile"));
-            if (index + 1 < arguments.size()) {
-                profileName = arguments.at(index + 1);
-            }
-        } else if (BrowserApp::captivePortal()) {
-            profileName = QStringLiteral("captiveportal");
+    const QStringList &arguments = QCoreApplication::arguments();
+    if (arguments.contains(QStringLiteral("-profile"))) {
+        int index = arguments.indexOf(QStringLiteral("-profile"));
+        if (index + 1 < arguments.size()) {
+            return arguments.at(index + 1);
         }
     }
-    return profileName;
+    return QStandardPaths::writableLocation(QStandardPaths::CacheLocation);
```

Comment:
That CacheLocation feels weird here. I would think that user profile should be in DataLocation even if there is some cache inside it. If it's really only about cache this would deserve a comment.

Reply:
There are all kind of things but yeah AppDataLocation sounds better.